### DISCUSSION TITLE
Can only one participant be created with the same domainid?

### DISCUSSION THREAD

**Discussion by zhuohuawang** [OpenDDS/OpenDDS#4280] [Q&A] [answered]
If I change the way the Publisher is created to look like this, then I find that subscribers can only receive one topic. Does this mean that only one participant can be created for the same domainid? 

In fact, I started Publisher this way from the very beginning, until I realized that subscribers could only receive data for one topic. Now I will try to correct the logical error you mentioned, but the result is still the same.

```
std::atomic_bool quit = false;
    for(size_t i =0;i < thrds.size();++i){
        std::thread t([&quit, i](){
            //Create a Publisher for each thread
            Publisher pub;
            std::string topic_name = "test_topic_" + std::to_string(i+1);
            uint32_t domian_id = 10;
            pub.create_publisher(domian_id,topic_name);
            Messenger::Message message;
            message.text = "sdafgdfasdasda";
            for(int index = 0;(index < 100000 && !quit);++index){
                message.index = index;
                message.timestamp = getTimestamp();
                pub.publish(domian_id,topic_name,true,message);
                std::this_thread::sleep_for(std::chrono::milliseconds(5 * 1000));
            }
        });
        thrds[i].swap(t);
    }
```
The other code contexts are in another discussion area:
[[](url)](https://github.com/OpenDDS/OpenDDS/discussions/4257#discussioncomment-7053437)

**Comment by mitza-oci** [comment, marked as answer]
No, multiple domain participants can coexist and be in different domains.  Configuring this depends on the transport type you've selected, see https://github.com/OpenDDS/OpenDDS/issues/270
